centreon/src/Core/Resources/Infrastructure/Repository/DbReadResourceRepository.php

```diff
@@ -286,17 +286,15 @@ public function findResourcesByAccessGroupIds(ResourceFilter $filter, array $acc
     private function addResourceAclSubRequest(array $accessGroupIds): string
     {
         $orConditions = array_map(
-            fn(ResourceACLProviderInterface $provider) => '(' . $provider->getACLSubRequest() . ')',
+            fn (ResourceACLProviderInterface $provider) =>  $provider->buildACLSubRequest($accessGroupIds),
             iterator_to_array($this->resourceACLProviders)
         );
 
         if (empty($orConditions)) {
             throw new \InvalidArgumentException(_('You must provide at least one ACL provider'));
         }
 
-        $pattern = ' AND EXISTS (SELECT 1 FROM `:dbstg`.centreon_acl acl WHERE (%s) AND acl.group_id IN (%s) LIMIT 1)';
-
-        return sprintf($pattern, join(' OR ', $orConditions), join(', ', $accessGroupIds));
+        return sprintf(' AND (%s)', join(' OR ', $orConditions));
```

Comment:
not a blocking point (it was already there), but IMHO we should not use php alias.
Prefer usage of `implode` instead of `join`

Reply:
Done